### ISSUE TITLE
Only validate first 3 version number field for msi and wix installers as default

### ISSUE THREAD

**Issue by ChristianGalla** [microsoft/winget-cli#2743]
### Description of the new feature / enhancement

According to the [documentation](https://learn.microsoft.com/en-us/windows/win32/msi/productversion#remarks), "Windows Installer uses only the first three fields of the product version. If you include a fourth field in your product version, the installer ignores the fourth field". Therefore, without any additional configuration, winget should only compare the first 3 version fields, too.

### Proposed technical implementation details

If the version number of an msi or wix installer is compared to the version number in apps and features, only the first 3 version fields should be compared. If the fourth field is missing or differs, it should be handled as the same version.

Only if the AppsAndFeatures field in the manifest is set, the whole version number should be compared.

**Comment by denelon** [comment]
WinGet is looking at the "displayVersion" for installed packages when performing version comparison. The "packageVersion" in the manifest is looked at as the secondary option if the AppsAndFeatures entries aren't present. Are there any specific examples where this is causing trouble today?

**Comment by Trenly** [comment]
I would also add that if only the first three fields are considered, then users would potentially miss out on patch versions and updates that they may otherwise want

**Comment by ChristianGalla** [comment]
As already mentioned, for the Windows Installer version 1.2.3.4 is the same as 1.2.3.5.
So, when version 1.2.3.4 is already installed and the installer for version 1.2.3.5 is started, it will usually not update the installed version.
I don't think there is a reason, winget should be stricter than the installer, but I can add more background:

Although, not all programs are already using it, best practice is to use [Semantic Versioning](https://semver.org/), so only 3 version numbers (major.minor.patch) are used. This "marketing version" is used as Git release tag and as release name, for example on GitHub or other websites. Therefore, it should also be used as winget version.

On the other hand for example [.NET assembly versions](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assemblyversionattribute.-ctor?redirectedfrom=MSDN&view=net-7.0#System_Reflection_AssemblyVersionAttribute__ctor_System_String_) have 4 parts (major.minor.build.revision). It is possible to auto generate the build and revision part in a build depending on the current date and time. This may be used in an automated CI/CD-Pipeline and is the real version number of the program that is also listed in apps and features.

If I have overlooked something, it should still be possible to pin an exact version in the manifest using AppsAndFeaturesEntries / DisplayVersion, but for the normal use case, only the 3-part marketing version should be used anywhere.

If you are new to winget, the chance is high, you don't know that it is possible to set AppsAndFeaturesEntries / DisplayVersion in the manifest. So new package contributors may not use it, even if they should to, like I did in my first update [Pull Request](https://github.com/microsoft/winget-pkgs/pull/90039). Luckily @mdanish-kh took the time to carefully check my pull request and suggested the changes, but if in the future, there is more automation, this may slip and result in wrong behavior of winget.

Currently, you can only set AppsAndFeaturesEntries / DisplayVersion via manifest. The documentation is hard to find and wingetcreate doesn't has a command line parameter, to for example easily set the different version number from a release pipeline.

I am only talking about msi and wix installers. Of course, if using for example a custom exe installer, the whole version should always be compared, because the installer may also compare the whole version number. I haven't had a look at the behavior of other installers, like msix, but maybe some of them are also only comparing the first 3 version numbers.

**Comment by denelon** [comment]
Yes, we have more work to do around authoring and documenting the explicit behaviors. We're working on additional validation to detect the "AppsAndFeatures" data during validation so we can prompt the user in the PR to add/correct the values.
